### PR TITLE
Rewrite factories to work better with Typescript

### LINKED ISSUE #360: Tweak factories to work with TS

### PR DESCRIPTION
**Pull request by mythmon**
Fixes #360

### CODE REVIEW

**Comment by mythmon** [comment]
I'd like to merge this *after* #379 so that we don't try and ship it in v2.1.1.

**Comment by mythmon** [comment]
This is unlocked, and is ok to review and merge now.

File: package.json [hunk 1]
```diff
@@ -36,6 +36,7 @@
     "@babel/preset-typescript": "7.10.4",
     "@testing-library/jest-dom": "5.11.4",
     "@testing-library/react": "10.4.9",
+    "@types/faker": "^4.1.12",
```
Comment (rehandalal) [inline package.json]: nit: pin this

**Review by rehandalal — APPROVED** [review]
phew! this looks good to me. 

r+wc

**Comment by mythmon** [comment]
bors r= @rehandalal

**Pull request merged**